tests/backwards_compatibility_tests/test_general_compatibility.py

```diff
@@ -0,0 +1,198 @@
+
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.client = marqo.Client(**cls.client_settings)
+        cls.logger.debug(f"Creating indexes {cls.indexes_to_test_on}")
+        cls.create_indexes(cls.indexes_to_test_on)
+        cls.indexes_to_delete = [cls.structured_index_metadata['indexName'],
+                                 cls.unstructured_index_metadata['indexName']]
+
+    def prepare(self):
+        """
+        Prepare the indexes and add documents for the test.
+        Also store the search results for later comparison.
+        """
+        with self.subTest("Adding documents to structured and unstructured indexes"):
+            try:
+                self.logger.debug(f'Feeding documents to {self.indexes_to_test_on}')
+                for index in self.indexes_to_test_on:
+                    if index.get("type") is not None and index.get('type') == 'structured':
+                        self.client.index(index_name=index['indexName']).add_documents(documents=self.docs)
+                    else:
+                        self.client.index(index_name=index['indexName']).add_documents(documents=self.docs,
+                                                                                       mappings=self.mappings,
+                                                                                       tensor_fields=self.tensor_fields)
+            except Exception as e:
+                self.logger.error(f"Exception occurred while adding documents {e}")
+
+        all_results = {}
+        # Loop through queries, search methods, and result keys to populate unstructured_results
+        with self.subTest("Storing search results to be compared later"):
+            for index in self.indexes_to_test_on:
+                index_name = index['indexName']
+                all_results[index_name] = {}
+
+                # For each index, store results for different search methods
+                for query, search_method, result_key in zip(self.queries, self.search_methods, self.result_keys):
+                    if index.get("type") is not None and index.get("type") == 'structured':
+                        if search_method == 'HYBRID':
+                            result = self.client.index(index_name).search(q=query, search_method=search_method, hybrid_parameters=self.hybrid_search_params)
+                        else:
+                            result = self.client.index(index_name).search(q=query, search_method=search_method, searchable_attributes=self.searchable_attributes[search_method])
+                    else:
+                        result = self.client.index(index_name).search(q=query, search_method=search_method)
+                    all_results[index_name][result_key] = result
+
+        # store the result of search across all structured & unstructured indexes
+        self.save_results_to_file(all_results)
+
+    def test_search(self):
```

Comment:
I think the organization should be more intuitive. General compatibility doesn't tell me much really. You could start with the APIs and have a folder for search tests for example, or use the documentation's structure (though I personally prefer the former)

Reply:
I would be better able to organise once I have a wholistic view of what all test cases to write, so I suggest that we do this in the next PR when I improve test coverage, let's let this be here for now